public/lib/validate.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Definitions:
+ * - NODE_OBJ:
+ *   { name: <string> }
+ * - LINK_OBJ:
+ *   {
+ *     source: <number>,
+ *     target: <number>,
+ *     value: <number>
+ *   }
+ * - HASHMAP: <number><object>
+ *   <number>,
+ *   {
+ *     source: <string>,
+ *     target: <string>,
+ *     value: number
+ *   }
+ */
+module.exports = function() {
+
+  /**
+   * Get node name of current index.
+   *
+   * @param {array<NODE_OBJ>} nodes - List of node objects.
+   * @param {number} name - The node index.
+   * @returns {string} - Current node name.
+   */
+  const _getNodeName = (nodes, index) => nodes[index].name;
+
+  /**
+   * Gets node index of current name.
+   *
+   * @param {array<NODE_OBJ>} nodes - List of node objects.
+   * @param {string} name - The node name.
+   * @returns {number} - Current node index. Returns `-1` if node name not found.
+   */
+  const _getNodeId = (nodes, name) => nodes.findIndex(node => node.name === name);
```

Comment:
I would propose to call this _getNodeIndex, because it is more specific. An id could be anything.